include/battle.h

```diff
@@ -831,6 +831,8 @@ struct BattleStruct
     u8 padding:4;
     u8 usedEjectItem;
     u8 usedMicleBerry;
+    u8 monCausingSleepClause[NUM_BATTLE_SIDES]; // Stores which pokemon on a given side is causing Sleep Clause to be active as the mon's index in the party
+    u8 sleepClauseEffectExempt[MAX_BATTLERS_COUNT]; // Stores whether effect should be exempt from triggering Sleep Clause (Effect Spore)
```

Comment:
1. What happens if a mon that has the except set switches out and what is it needed for. It isn't very clear from the code?
2. This can be a bitfield instead.

Reply:
Nothing relevant happens when the mon switches actually. This is set to true or false per sleep move used against a specific battler, and also for effects that don't trigger sleep. Yawning your partner, Effect Spore, etc.

In code, it's to flag anything that ends up using the `SetMoveEffect` function to trigger sleep that shouldn't activate sleep clause. The reason switch outs don't matter is that's it's reset for every sleep effect or move used, so the previous state doesn't matter. The only instance in which a previous turns stat matters is the case of Yawning your partner, where if nothing occurs it will still block yawn from activating clause, and if some other sleep effect occurs targeting your partner that effect can overwrite this check so the functionality still works.

I hope that makes sense lol

Will convert to bitfield :)